Makefile

```diff
@@ -243,28 +235,19 @@ uninstall:
 
 uninstall-lite:
 	if test -f nxcomp/Makefile; then ${MAKE} -C nxcomp $@; fi
+	if test -f nxproxy/Makefile; then ${MAKE} -C nxproxy $@; fi
 
-	# uninstall nproxy wrapper script
-	$(RM_FILE) $(DESTDIR)$(BINDIR)/nxproxy
-	# FIXME: don't use uninstall rule in nxproxy/Makefile.in, let's do
-        # it on our own for now...
-	$(RM_FILE) $(DESTDIR)$(NXLIBDIR)/bin/nxproxy
-	$(RM_DIR) $(DESTDIR)$(NXLIBDIR)/bin/
-	$(RM_FILE) $(DESTDIR)$(PREFIX)/share/man/man1/*.1
 	$(RM_FILE) $(DESTDIR)$(PREFIX)/share/nx/VERSION.nxproxy
 	$(RM_DIR) $(DESTDIR)$(PREFIX)/share/nx/
 
 uninstall-full:
-	for f in nxagent; do \
-	    $(RM_FILE) $(DESTDIR)$(BINDIR)/$$f; done
+	if test -f nxcompshad/Makefile; then ${MAKE} -C nxcompshad $@; fi
+	if test -f nx-X11/lib/Makefile; then ${MAKE} -C nx-X11/lib $@; fi
+
+	$(RM_FILE) $(DESTDIR)$(BINDIR)/nxagent
 
 	$(RM_FILE) $(DESTDIR)$(PREFIX)/share/nx/VERSION.nxagent
 	$(RM_DIR) $(DESTDIR)$(PREFIX)/share/nx/
 
-	if test -d nx-X11; then \
-	    if test -f nxcompshad/Makefile; then ${MAKE} -C nxcompshad $@; fi; \
-	    if test -f nx-X11/Makefile; then \
-	        if test -d $(NXLIBDIR); then rm -rf $(NXLIBDIR); fi; \
```

Comment:
This is actually a bug here as well, since we didn't respect `DISTDIR` for `libX11` either.

Reply:
> I added DESTDIR here now in a to-be-rebased commit.

Okay!